File: 2025/day2/src/main.rs
use std::fs;

fn main() -> std::io::Result<()> {
    let mut answer: u128 = 0;
    let content = fs::read_to_string("input")?;

    for item in content
        .split(',')
        .map(|s| s.trim())
        .filter(|s| !s.is_empty())
    {
        if let Some((x_str, y_str)) = item.split_once('-') {
            let mut first: u128 = x_str.parse().expect("Failed to parse first");
            let last: u128 = y_str.parse().expect("Failed to parse last");

            while first <= last {
                if is_invalid(&first) {
                    answer += first;
                    println!("{first}")
                }
                first += 1;
            }
        }
    }
    println!("Solution:{answer}");
    Ok(())
}

fn is_invalid(number: &u128) -> bool {
    let number_str: String = number.to_string();
    let num_len = number_str.len();

    if num_len <= 1 {
        return false;
    }

    for chunk_size in 1..=(num_len / 2) {
        if !num_len.is_multiple_of(chunk_size) {
            continue;
        }

        let chunk: String = number_str.chars().take(chunk_size).collect();
        if number_str == chunk.repeat(num_len / chunk_size) {
            return true;
        }
    }

    false
}

File: 2025/day2/src/main.ts
const fs = require("fs");
const path = require("path");

function main(): number {
  let answer: number = 0;
  const inputPath = path.join(__dirname, "..", "input");
  const fileContents = fs.readFileSync(inputPath, "utf8");

  const input: string[] = fileContents
    .split(",")
    .map((s: string) => s.trim())
    .filter(Boolean);

  for (let i = 0; i < input.length; i++) {
    const [first, last] = input[i].split("-");

    let firstNum: number = Number(first);
    let lastNum: number = Number(last);

    while (firstNum <= lastNum) {
      if (isInvalid(firstNum)) {
        answer += firstNum;
        console.log(firstNum);
      }

      firstNum++;
    }
  }

  return answer;
}

function isInvalid(num: number): boolean {
  const numStr: string = num.toString();
  const numLen: number = numStr.length;

  if (numLen <= 1) return false;

  for (let chunkSize = 1; chunkSize <= numLen / 2; chunkSize++) {
    if (numLen % chunkSize !== 0) continue;

    const chunk: string = numStr.slice(0, chunkSize);

    if (numStr === chunk.repeat(numLen / chunkSize)) return true;
  }

  return false;
}

function isInvalidRegex(num: number): boolean {
  const numStr = num.toString();
  return /^(.+?)\1+$/.test(numStr);
}

const x = main();

console.log(`solution:${x}`);
